apps/web/src/components/message-center-handler/index.tsx

```diff
@@ -0,0 +1,35 @@
+import { toast } from '@affine/component';
+import { getApis, MessageCenter } from '@affine/datacenter';
+import { useRouter } from 'next/router';
+import { useEffect } from 'react';
+
+export function MessageCenterHandler({
+  children,
+}: {
+  children?: React.ReactNode;
+}) {
+  const router = useRouter();
+  useEffect(() => {
+    const instance = MessageCenter.getInstance();
+    if (instance) {
+      return instance.onMessage(async message => {
+        if (message.code === MessageCenter.messageCode.noPermission) {
+          // todo: translate message
+          // todo: more specific message for accessing different resources
+          // todo: error toast style
+          toast('You have no permission to access this workspace');
+          getApis().auth.clear();
+          // the status of the app right now is unknown, and it won't help if we let
```

Comment:
use error boundary

Reply:
IMO error boundary is only for unexpected exceptions that could crash an app. We could use it to report errors or provide a last-resort error screen to let the user do some actions manually. Handling 401 errors are not in this case.
If we know the reason for the error & know how to deal with it, error boundary should be avoided.